File: 0043-multiply-strings/0043-multiply-strings.py
class Solution:
    def multiply(self, num1: str, num2: str) -> str:
        mult=1
        count=0
        num1=[*num1]
        num2=int(num2)
        for i in range(len(num1)-1,-1,-1):
            count+=int(num1[i])*num2*mult
            mult*=10
        return str(count)

File: 0043-multiply-strings/0043-multiply-strings.cpp
class Solution {
public:
    string multiply(string num1, string num2) {
        long long m1 = 1, ans = 0, carry = 0;
        
        // Loop over num1 from the last digit
        for (int i = num1.size() - 1; i >= 0; i--) {
            long long m2 = 1, temp = 0;
            carry = 0;
            for (int j = num2.size() - 1; j >= 0; j--) {
                int curr = (num2[j] - '0') * (num1[i] - '0') + carry;
                carry = curr / 10;
                int add = (curr % 10) * m2;
                temp += add;
                m2 *= 10;
            }
            if (carry != 0) {
                temp += carry * m2;
            }
            ans += temp * m1;
            m1 *= 10;
        }
        return to_string(ans);
    }
};
